lib/actions/field-trip.js

```diff
@@ -0,0 +1,424 @@
+    queryData.append('note.note', note.note)
+    queryData.append('note.type', note.type)
+    queryData.append('note.userName', username)
+    queryData.append('requestId', request.id)
+    return fetch(`${datastoreUrl}/fieldtrip/addNote`,
+      {method: 'POST', body: queryData}
+    )
+      .then(() => dispatch(fetchFieldTripDetails(request.id)))
+      .catch(err => {
+        alert(`Error adding field trip note: ${JSON.stringify(err)}`)
+      })
+  }
+}
+
+/**
+ * Delete a specific note for a field trip request.
+ */
+export function deleteFieldTripNote (request, noteId) {
+  return function (dispatch, getState) {
+    const {callTaker, otp} = getState()
+    const {datastoreUrl} = otp.config
+    if (sessionIsInvalid(callTaker.session)) return
+    const {sessionId} = callTaker.session
+    const queryData = new FormData()
+    queryData.append('sessionId', sessionId)
+    queryData.append('noteId', noteId)
+    return fetch(`${datastoreUrl}/fieldtrip/deleteNote`,
+      {method: 'POST', body: queryData}
+    )
+      .then(() => dispatch(fetchFieldTripDetails(request.id)))
+      .catch(err => {
+        alert(`Error deleting field trip note: ${JSON.stringify(err)}`)
+      })
+  }
+}
+
+// TODO: Enable processPlan when planning/saving trip for field trip request.
+// function processPlan (tripPlan, restoring) {
+//   if (updateActiveOnly) {
+//     var itinIndex = itinWidget.activeIndex
+//     tripPlan.itineraries[0].groupSize = groupPlan.itineraries[itinIndex].groupSize
+//     itinWidget.updateItineraries(tripPlan.itineraries)
+//     updateActiveOnly = false
+//     drawItinerary(tripPlan.itineraries[0])
+//     return
+//   }
+//
+//   if (groupPlan == null) {
+//     groupPlan = new otp.modules.planner.TripPlan(null, _.extend(tripPlan.queryParams, { groupSize : groupSize }))
+//   }
+//
+//   if (itinWidget == null) createItinerariesWidget()
+//
+//   var itin = tripPlan.itineraries[0]
+//   var capacity = itin.getGroupTripCapacity()
+//
+//   // if this itin shares a vehicle trip with another one already in use, only use the remainingCapacity (as set in checkTripValidity())
+//   if (itinCapacity) capacity = Math.min(capacity, itinCapacity)
+//
+//   groupPlan.addItinerary(itin)
+//
+//   var transitLegs = itin.getTransitLegs()
+//   for (var i = 0; i < transitLegs.length; i++) {
+//     var leg = transitLegs[i]
+//     bannedSegments.push({
+//       tripId : leg.tripId,
+//       fromStopIndex : leg.from.stopIndex,
+//       toStopIndex : leg.to.stopIndex,
+//     })
+//   }
+//
+//   setBannedTrips()
+//
+//   if (currentGroupSize > capacity) {
+//     // group members remain. plan another trip
+//     currentGroupSize -= capacity
+//     itin.groupSize = capacity
+//     //console.log("remaining: "+currentGroupSize)
+//     itinCapacity = null
+//     planTrip()
+//   } else {
+//     // we're done. show the results
+//     itin.groupSize = currentGroupSize
+//     showResults()
+//   }
+// }
```

Comment:
I'm generally not a fan of leaving in large commente-out code blocks. If this isn't used currently, can it be removed for now?

Reply:
Initially I had planned to include this functionality, but it ended up being a bigger lift than expected and I didn't want the PR to grow too large. I did some work adapting the old OTP field trip module code here, so I didn't want to totally yank it out. I suppose I could remove this stuff and just store on my local disk? Keeping it under version control in the repo seemed like a better alternative though.

Reply:
One thing that would really help with all this is to have a roadmap/issue to reference where we can explicitly point to this kind of stuff being needed in the future. Otherwise, my gut reaction is: "why the heck is this here?". So if this is really needed and will be uncommented to live soon, reference an issue or roadmap or something.